src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1679,6 +1667,22 @@ public T next() {
 
     abstract class AbstractIterator<T> implements Iterator<T> {
 
+        public boolean equals(Object o) {
```

Comment:
This is important fix. It was surprise for me, but Object.equals() overrides and hides Iterator.equals() method. We should implement it in class but not in interface.